code/datums/components/animation.dm

```diff
@@ -0,0 +1,70 @@
+/datum/component/animation
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	var/list/animations				// Assoc List of animation_objects paired to their signal list(signal = animation_object)
+	var/list/random_animations		// Weighted list of animations. list(animation_object = weight)
+	var/random_animation_min_delay	// How long at a minimum in between random animations
+	var/random_animation_max_delay	// How long at a maximum in between random animations
+	var/animation_endings = list()	// Prevents animations from disrupting one another. It is an assoc list with the owner as key and the end time as value
+
+
+/* This component will flick animations on objects defined in the input
+ * animation_list assoc list of animations with the signal as key
+ * random_animation_list weighted list of animations
+ * random_animation_min_delay the minimum delay between random animations
+ * random_animation_max_delay the maximum delay between random animations
+ */
+/datum/component/animation/Initialize(list/datum/animation_object/animation_list, list/datum/animation_object/random_animation_list, random_animation_min_delay, random_animation_max_delay)
+	if(!isatom(parent))
+		log_debug("/datum/component/animation initialized with non atom parent: [parent] ([parent.type])")
+		return COMPONENT_INCOMPATIBLE
+
+	if(!length(animation_list) && !length(random_animation_list))
+		log_debug("/datum/component/animation initialized without animations: [parent] ([parent.type])")
+		return COMPONENT_INCOMPATIBLE
+
+	if(length(random_animation_list) && (isnull(random_animation_min_delay) || isnull(random_animation_max_delay)))
+		log_debug("/datum/component/animation initialized with random animations without proper timing : [parent] ([parent.type]), ([random_animation_min_delay] - [random_animation_max_delay])")
+		return COMPONENT_INCOMPATIBLE
+
+	if(length(animation_list))
+		animations = animation_list.Copy()
+		for(var/signal in animation_list)
+			if(!QDELETED(animation_list[signal])) // Invalid animation_objects will be filtered
+				RegisterSignal(parent, signal, CALLBACK(src, .proc/flick_animation, animation_list[signal], signal))
+
+	if(length(random_animation_list))
+		src.random_animations = random_animation_list.Copy()
+		src.random_animation_min_delay = random_animation_min_delay
+		src.random_animation_max_delay = random_animation_max_delay
```

Comment:
To quote the prior maintainer DZD.

"if you have to use implied `src` with your arguments, then you should rename your arguments"